orchagent/isolationgrouporch.cpp

```diff
@@ -0,0 +1,754 @@
+        doIsoGrpTblTask(consumer);
+    }
+    else
+    {
+        SWSS_LOG_ERROR("Invalid table %s", table_name.c_str());
+    }
+}
+
+void
+IsoGrpOrch::doIsoGrpTblTask(Consumer &consumer)
+{
+    SWSS_LOG_ENTER();
+    isolation_group_status_t status = ISO_GRP_STATUS_SUCCESS;
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
+        string op = kfvOp(t);
+        string key = kfvKey(t);
+
+        size_t sep_loc = key.find(consumer.getConsumerTable()->getTableNameSeparator().c_str());
+        string name = key.substr(0, sep_loc);
+
+        SWSS_LOG_DEBUG("Op:%s IsoGrp:%s", op.c_str(), name.c_str());
+
+        if (op == SET_COMMAND)
+        {
+            isolation_group_type_t type = ISOLATION_GROUP_TYPE_INVALID;
+            string descr("");
+            string bind_ports("");
+            string mem_ports("");
+
+            for (auto itp : kfvFieldsValues(t))
+            {
+                string attr_name = to_upper(fvField(itp));
+                string attr_value = fvValue(itp);
+
+                if (attr_name == ISOLATION_GRP_DESCRIPTION)
+                {
+                    descr = attr_value;
+                }
+                else if (attr_name == ISOLATION_GRP_TYPE)
+                {
+                    if (ISOLATION_GRP_TYPE_PORT == attr_value)
+                    {
+                        type = ISOLATION_GROUP_TYPE_PORT;
+                    }
+                    else if (ISOLATION_GRP_TYPE_BRIDGE_PORT == attr_value)
+                    {
+                        type = ISOLATION_GROUP_TYPE_BRIDGE_PORT;
+                    }
+                }
+                else if (attr_name == ISOLATION_GRP_PORTS)
+                {
+                    bind_ports = attr_value;
+                }
+                else if (attr_name == ISOLATION_GRP_MEMBERS)
+                {
+                    mem_ports = attr_value;
+                }
+            }
+
+            status = addIsolationGroup(name, type, descr, bind_ports, mem_ports);
+            if (ISO_GRP_STATUS_SUCCESS == status)
+            {
+                auto grp = getIsolationGroup(name);
+                if (!grp->isObserver(this))
+                {
+                    IsolationGroupUpdate update = {grp.get(), true};
+                    grp->notifyObservers(SUBJECT_TYPE_ISOLATION_GROUP_CHANGE, &update);
+
+                    grp->attach(this);
+                }
+            }
+        }
+        else
+        {
+            auto grp = getIsolationGroup(name);
+            if (grp)
+            {
+                grp->detach(this);
+
+                /* Send a notification and see if observers want to detach */
+                IsolationGroupUpdate update = {grp.get(), false};
+                grp->notifyObservers(SUBJECT_TYPE_ISOLATION_GROUP_CHANGE, &update);
+            }
+
+            // Finally delete it if it
+            status = delIsolationGroup(name);
```

Comment:
Looks like `delIsolationGroup()` should be moved under `if (grp)` as well.

Reply:
Thanks for the comment, as the If (grp) doenst return eventually the delIsolationGroup() is called.

Reply:
Yes, but if `grp == NULL` it means nothing to delete. Right? So, we should not call `delIsolationGroup()` in this case.